shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -1449,6 +1449,13 @@ public void removeFlutterEngineAttachmentListener(
         .send();
   }
 
+  private FlutterViewDelegate delegate = new FlutterViewDelegate();
+
+  @VisibleForTesting
+  public void setDelegate(@NonNull FlutterViewDelegate delegate) {
+    this.delegate = delegate;
+  }
+
   private void sendViewportMetricsToFlutter() {
```

Comment:
Looking at the design of this class I am seeing a pattern of the flutterView having methods that trigger on different android lifecycle events, Then ViewportMetrics being updated then at the end of that triggering method calling `sendViewportMetricsToFlutter`. 

Looking at this code we do the calculation every send. When I was searching internally I did not find documentation of a lifecycle method to trigger on so maybe this is moot. If this code moved to onApplyWindowInsets would it still work?